src/Control/CLIRequestBuilder.php

```diff
@@ -66,4 +68,19 @@ public static function cleanEnvironment(array $variables)
         // Parse rest of variables as standard
         return parent::cleanEnvironment($variables);
     }
+
+    /**
+     * @param array $variables
+     * @param string $input
+     * @return HTTPRequest
+     */
+    public static function createFromVariables(array $variables, $input)
+    {
+        $request = parent::createFromVariables($variables, $input);
+        // unset scheme so that SS_BASE_URL can provide `is_https` information if required
+        $scheme = parse_url(Environment::getEnv('SS_BASE_URL'), PHP_URL_SCHEME);
+        $request->setScheme($scheme);
```

Comment:
This really should have an `if` around it. Not all base_urls have schema set; They could be protocol-less absolute url. (e.g. `//localhost`). Even that case notwithstanding, you should avoid setting a null scheme in case of error.

Reply:
Patch here: https://github.com/silverstripe/silverstripe-framework/pull/8358